core/src/style/style.cpp

```diff
@@ -68,57 +75,104 @@ void Style::setLightingType(LightingType _type){
 
 }
 
-void Style::addLayer(const std::pair<std::string, StyleParamMap>&& _layer) {
+void Style::addLayer(std::shared_ptr<SceneLayer> _layer) {
 
     m_layers.push_back(std::move(_layer));
 
 }
 
+void Style::applyLayerFiltering(const Feature& _feature, const Context& _ctx, std::bitset<MAX_LAYERS>& _uniqueID,
+                                   StyleParamMap& _styleParamMapMix, std::weak_ptr<SceneLayer> _uberLayer) const {
+
+    std::vector<std::weak_ptr<SceneLayer>> sLayers;
+    sLayers.reserve(_uberLayer.lock()->getSublayers().size() + 1);
```

Comment:
What's the benefit of a `weak_ptr` if you're not going to check the result of `lock`?